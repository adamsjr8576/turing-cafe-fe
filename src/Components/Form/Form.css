.reservation-form {
  width: 90%;
  display: flex;
  align-items: center;
  justify-content: space-around;
}

.form-input {
  padding: 10px;
  width: 15%;
  font-size: 1em;
  border-radius: 10px;
  border: 0;
}

.reserve-button {
  width: 12%;
  padding: 10px;
  font-size: 1em;
  border-radius: 10px;
  cursor: pointer;
}

.reserve-button:hover {
  background-color: lightblue;
}
