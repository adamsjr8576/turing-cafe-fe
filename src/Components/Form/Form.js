import React, { Component } from 'react';
import './Form.css';

class Form extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      date: '',
      time: '',
      number: 0
    }
  }

  updateInputState = (event) => {
    this.setState({ [event.target.name]: event.target.value })
  }

  clearState = () => {
    this.setState({ name:'', date:'', time:'', number:0 })
  }

  handleReservationPost = () => {
    const { name, date, time, number } = this.state;
    const dateSplit = date.split('-');
    dateSplit.shift();
    const newDate = dateSplit.join('/');
    const options = {
      method: "POST",
      body: JSON.stringify({
        name: name,
        date: newDate,
        time: time,
        number: parseInt(number)
      }),
      headers: {
        "Content-Type": "application/json"
        }
      }
    fetch('http://localhost:3001/api/v1/reservations', options)
    .then(res => res.json())
    .then(data => this.props.addReservation(data))
    .catch(err => console.log(err))
  }



  render() {
    const { name, date, time, number } = this.state;
    return (
      <form className='reservation-form'>
        <input
          type='text'
          name='name'
          value={name}
          placeholder='Name'
          className='form-input input1'
          onChange={(event) => this.updateInputState(event)}
        />
        <input
          type='date'
          name='date'
          value={date}
          placeholder='Check-in Date'
          className='form-input input2'
          onChange={(event) => this.updateInputState(event)}
          />
        <input
          type='text'
          name='time'
          value={time}
          placeholder='Check-In Time'
          className='form-input input3'
          onChange={(event) => this.updateInputState(event)}
        />
        <input
          type='number'
          name='number'
          value={number}
          placeholder='Number of Guests'
          className='form-input input4'
          onChange={(event) => this.updateInputState(event)}
        />
        <button type='button' className='reserve-button' onClick={() => {this.handleReservationPost(); this.clearState()}}>Make Reservation</button>
      </form>
    )
  }
}

export default Form;
