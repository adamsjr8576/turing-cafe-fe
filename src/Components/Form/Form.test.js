import React from 'react';
import ReactDOM from 'react-dom';
import Form from './Form';
import { shallow } from 'enzyme';

describe('Form', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Form
      addReservation={jest.fn()}
      />);
  });

  it('should match the snapshot with all data passing correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state with the input value when updateInputState is called', () => {
      const mockEvent = {target: {name:'name', value:'John Adams'}}

      expect(wrapper.state('name')).toEqual('');

      wrapper.instance().updateInputState(mockEvent);

      expect(wrapper.state('name')).toEqual('John Adams');
  });

  it('should call updateInputState onchange of the input', () => {
    wrapper.instance().updateInputState = jest.fn();
    wrapper.instance().forceUpdate();
    const mockEvent = {target: {name:'name', value: 'John'}}

    wrapper.find('.input1').simulate('change', mockEvent);
    expect(wrapper.instance().updateInputState).toHaveBeenCalledWith(mockEvent);
  })

});
