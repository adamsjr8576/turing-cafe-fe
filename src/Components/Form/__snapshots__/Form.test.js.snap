// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot with all data passing correctly 1`] = `
<form
  className="reservation-form"
>
  <input
    className="form-input input1"
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    className="form-input input2"
    name="date"
    onChange={[Function]}
    placeholder="Check-in Date"
    type="date"
    value=""
  />
  <input
    className="form-input input3"
    name="time"
    onChange={[Function]}
    placeholder="Check-In Time"
    type="text"
    value=""
  />
  <input
    className="form-input input4"
    name="number"
    onChange={[Function]}
    placeholder="Number of Guests"
    type="number"
    value={0}
  />
  <button
    className="reserve-button"
    onClick={[Function]}
    type="button"
  >
    Make Reservation
  </button>
</form>
`;
